addon/mixins/authenticated-route-mixin.js

```diff
@@ -61,6 +61,7 @@ export default Mixin.create({
     @public
   */
   session: service('session'),
+  router: service('router'),
```

Comment:
I agree w/ @marcoow's concern

> The problem with this is that it only works if the router service exists. 

For versions of Ember that predate the router service, you can get ahold of the router singleton directly via `getOwner(this).lookup('router:main')`

Reply:
Updated to use router service or get the router singleton.